sys/auto_init/saul/auto_init_pwm.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2017 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+/*
+ * @ingroup     auto_init_saul
+ * @{
+ *
+ * @file
+ * @brief       Auto initialization of PWM pins directly mapped to SAUL reg
+ *
+ * @author      Francisco Acosta <francisco. acosta@inria.fr>
+ *
+ * @}
+ */
+
+#ifdef MODULE_SAUL_PWM
+
+#include "log.h"
+#include "saul_reg.h"
+#include "saul/periph.h"
+#include "pwm_params.h"
+#include "periph/pwm.h"
+
+/**
+ * @brief   Define the number of configured sensors
+ */
+#define SAUL_PWM_NUMOF    (sizeof(saul_pwm_params)/sizeof(saul_pwm_params[0]))
+
+/**
+ * @brief   Allocate memory for the device descriptors
+ */
+static const saul_pwm_params_t *saul_pwms[SAUL_PWM_NUMOF];
```

Comment:
no need for this buffer, we could reference the parameters defined by the board directly (see #7586).